### DISCUSSION TITLE
DGS Java client

### DISCUSSION THREAD

**Discussion by mparthasarathi** [Netflix/dgs-framework#1055] [Q&A]
From the [docs](https://netflix.github.io/dgs/advanced/java-client/#type-safe-query-api)
> The GraphQLQueryRequest is a class from graphql-dgs-client

But I don't see any class named `GraphQLQueryRequest` in `graphql-dgs-client-4.10.2.jar` and it's present in `graphql-dgs-codegen-client-core-5.1.17.jar` which forces this dependency to be present in compile/runtime and doesn't sound right.

Am I missing something here?

**Comment by berngp** [comment]
We moved the`GraphQLQueryRequest` class to the DGS Codegen project since it is really meant to be used when leveraging the [DGS Codgen Plugin](https://github.com/Netflix/dgs-codegen). This allowed us to decouple the release of the framework from that of the plugin.

**Reply by Tr1monster** [reply]
I would be great if you can update your documentation at [this](https://netflix.github.io/dgs/advanced/java-client/#:~:text=The%20GraphQLQueryRequest%20is%20a%20class%20from%20graphql%2Ddgs%2Dclient) location. It takes me only a few minutes to find this thread but a correct documentation would be awesome.

**Reply by berngp** [reply]
Thanks for the reminder @Tr1monster
